src/test_workflow/integ_test/topology.py

```diff
@@ -0,0 +1,187 @@
+            for cluster_manager_node in cluster.cluster_manager_nodes:
+                cluster_manager_nodes_endpoints.append(NodeEndpoint(cluster_manager_node))
+
+            topology_endpoints_list.append(ClusterEndpoint(cluster_config.cluster_name, data_node_endpoints, cluster_manager_nodes_endpoints))
+            total_nodes_in_cluster = cluster_config.data_nodes + cluster_config.cluster_manager_nodes
+            current_port += total_nodes_in_cluster
+            current_work_dir_num += total_nodes_in_cluster
+            current_transport_port += total_nodes_in_cluster
+        return topology_endpoints_list
+
+    def destroy(self) -> None:
+        for topology_cluster in self.clusters:
+            for datanode in topology_cluster.data_nodes:
+                datanode.cluster.terminate()
+            for managernode in topology_cluster.cluster_manager_nodes:
+                managernode.cluster.terminate()
+        self.clusters = []
+
+
+class DataNode:
+    def __init__(
+        self,
+        dependency_installer: DependencyInstaller,
+        work_dir: str,
+        component_name: str,
+        additional_cluster_config: dict,
+        bundle_manifest: BundleManifest,
+        security_enabled: bool,
+        component_test_config: str,
+        test_recorder: TestRecorder,
+        current_transport_port: int,
+        cluster_port: int = 9200
+    ):
+        self.cluster = LocalTestCluster.create_cluster(
+            dependency_installer,
+            work_dir,
+            component_name,
+            additional_cluster_config,
+            bundle_manifest,
+            security_enabled,
+            component_test_config,
+            test_recorder,
+            cluster_port)
+        self.port = cluster_port
+        self.transport = current_transport_port
+
+
+class NodeEndpoint:
+    def __init__(self, *args: Any):
+        if len(args) == 1:
+            self.endpoint = args[0].cluster.endpoint
+            self.port = args[0].port
+            self.transport = args[0].transport
+        else:
+            self.endpoint = args[0]
+            self.port = args[1]
+            self.transport = args[2]
+
+
+class TopologyTestCluster:
+    def __init__(
+        self,
+        cluster_name: str,
+        cluster_config: ClusterConfig,
+        current_port: int,
+        current_transport_port: int,
+        current_work_dir_num: int,
+        additional_cluster_config: dict,
+        component_name: str,
+        bundle_manifest: BundleManifest,
+        security_enabled: bool,
+        component_test_config: str,
+        test_recorder: TestRecorder,
+        dependency_installer: DependencyInstaller,
+        work_dir: Path
+    ):
+        self.cluster_name = cluster_name
+        self.data_nodes: list = []
+        self.cluster_manager_nodes: list = []
+        additional_cluster_config['cluster.name'] = "opensearch" + cluster_config.cluster_name
+
+        for i in range(cluster_config.data_nodes):
```

Comment:
Is there any limitation on the number of nodes to be added?
Ex: 9200+100 will be 9300 so I assume the limit is 99?
Thanks.

Reply:
yes, total number of nodes is 100

Reply:
Hi @monu-aws do you have a limit already setup in your code?
Or you are expecting people to have good intention and dont add more than 100 nodes?

Reply:
Added this assertion.